### DISCUSSION TITLE
3.0 integration into existing NextJS website

### DISCUSSION THREAD

**Discussion by asjustis** [payloadcms/payload#8890] [General]
I am quite struggling with integrating the beta version into my existing project. And there seems to be info about that, but I get into more questions than answers reading it all. Could somebody shed some light on it? 

It looks like a recommended way is to "just run a one single-line command" `npx create-payload-app@beta` to make it work, but it just introduces a new project to a selected folder. If I add it to a separate folder for Payload, it exists as a separate project, if I put it into my website's page, it goes on to override pretty much everything that Payload needs (`package.json`, configs, etc.). So I wonder how to correctly add the Payload to the existing website. 

From the demo app, it shows ```Contents from src/app will have to be moved into a new directory src/app/(app) so that Payload's root layout and routes can remain isolated from the rest of your app.``` but it doesn't reflect all the Payload project files, etc.

Thanks for any help!

**Comment by denolfe** [comment]
Hey @asjustis, the `create-payload-app@beta` command will do 2 different things depending on where you run it.

- If you're _inside of an existing Next.js project_, it will prompt you accordingly and add the necessary Payload files
- If you're not inside a Next.js project, you'll be prompted for one of the existing templates that will get you going from scratch.

If inside of a fresh Next.js project, you'll need to make a small change to the folder layout in order to be compatible with Payload, this is the `src/app` prompt that you are seeing. This equates to moving everything that exists inside of `src/app` and moving it into a new directory called `src/app/(app)` or similar. Once you do this step, re-run the cli command, and it will be able to finish the process.

Hope that makes sense.
